monai/bundle/scripts.py

```diff
@@ -116,6 +120,166 @@ def _get_fake_spatial_shape(shape: Sequence[Union[str, int]], p: int = 1, n: int
+        extractall(filepath=filepath, output_dir=bundle_dir_, has_base=True)
+    elif source_ == "github":
+        if name_ is None or repo_ is None:
+            raise ValueError(
+                f"To download from source: Github, `name` and `repo` must be provided, got {name_} and {repo_}."
+            )
+        _download_from_github(repo=repo_, download_path=bundle_dir_, filename=name_, progress=progress_)
+    else:
+        raise NotImplementedError(
+            f"Currently only download from provided URL in `url` or Github is implemented, got source: {source_}."
+        )
+
+
+def load(
+    name: str,
+    load_ts_module: bool = False,
+    bundle_dir: Optional[PathLike] = None,
+    source: str = "github",
+    repo: Optional[str] = None,
+    progress: bool = True,
+    device: str = "cpu",
+    net_name: Optional[str] = None,
+    **net_kwargs,
+):
+    """
+    Load model weights or TorchScript module of a bundle.
+    If the weights file does not exist locally, it will be downloaded first.
+    The function can return weights, an instantiated network that loaded the weights, or a TorchScript module.
+
+    Args:
+        name: bundle name.
+        load_ts_module: a flag to specify if loading the TorchScript module.
+        bundle_dir: the directory the weights/TorchScript module will be loaded from.
+            Default is `bundle` subfolder under`torch.hub get_dir()`.
+        source: the place that saved the bundle.
+            If `source` is `github`, the bundle should be within the releases.
+        repo: the repo name. If the weights file does not exist locally and `url` is `None`, it must be provided.
+            If `source` is `github`, it should be in the form of `repo_owner/repo_name/release_tag`.
+            For example: `Project-MONAI/MONAI-extra-test-data/0.8.1`.
+        progress: whether to display a progress bar when downloading.
+        device: target device of returned weights or module.
+        net_name: if not `None`, a corresponding network will be instantiated and load the achieved weights.
+            This argument only works when loading weights.
+        net_kwargs: other arguments that are used to instantiate the network class defined by `net_name`.
+
+    """
+    if bundle_dir is None:
+        get_dir, has_home = optional_import("torch.hub", name="get_dir")
+        if has_home:
+            bundle_dir = Path(get_dir()) / "bundle"
+        else:
+            raise ValueError("bundle_dir=None, but no suitable default directory computed. Upgrade Pytorch to 1.6+ ?")
+    bundle_dir = Path(bundle_dir)
+
+    weights_name: str = "model.ts" if load_ts_module is True else "model.pt"
+    model_file_path = os.path.join(bundle_dir, name, weights_name)
+    if not os.path.exists(model_file_path):
+        download(name=name, bundle_dir=bundle_dir, source=source, repo=repo, progress=progress)
+
+    # loading with `torch.jit.load`
+    if load_ts_module is True:
```

Comment:
For `TorchScript` model, I would suggest to leverage `load_net_with_metadata ` to return `model`, `metadata`, `configs`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/torchscript_utils.py#L100
It's consistent with our `ckpt_export` API:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/scripts.py#L418
Seems you just need to add an optional arg `config_files` mapping to the `more_extra_files` arg.

What do you think?

Reply:
I will also enhance the `TorchScript` utils according to the discussion: https://github.com/Project-MONAI/MONAI/issues/4094#issuecomment-1105363600

Thanks.